tests/foreman/ui_airgun/test_contentview.py

```diff
@@ -1350,3 +1362,503 @@ def test_positive_update_filter_affected_repos(session, module_org):
+        4. Publish and promote the content view
+
+    :expectedresults:
+        1. Custom puppet module file successfully uploaded
+        2. Puppet module successfully added to content view
+        3. Content view successfully published and promoted
+
+    :BZ: 1335833
+
+    :CaseLevel: System
+    """
+    cv_name = gen_string('alpha')
+    env = entities.LifecycleEnvironment(organization=module_org).create()
+    # Creates new custom product via API's
+    product = entities.Product(organization=module_org).create()
+    # Creates new custom repository via API's
+    repo = entities.Repository(
+        url=FAKE_0_PUPPET_REPO,
+        content_type=REPO_TYPE['puppet'],
+        product=product,
+    ).create()
+    # Sync repo
+    call_entity_method_with_timeout(
+        entities.Repository(id=repo.id).sync, timeout=1500)
+
+    with session:
+        repo_values = session.repository.read(product.name, repo.name)
+        initial_modules_count = int(repo_values['content_counts']['Puppet Modules'])
+        session.repository.upload_content(
+            product.name, repo.name, get_data_file(PUPPET_MODULE_CUSTOM_FILE_NAME))
+        repo_values = session.repository.read(product.name, repo.name)
+        assert (
+            int(repo_values['content_counts']['Puppet Modules'])
+            == initial_modules_count + 1
+        )
+        session.contentview.create({'name': cv_name})
+        assert session.contentview.search(cv_name)[0]['Name'] == cv_name
+        session.contentview.add_puppet_module(cv_name, PUPPET_MODULE_CUSTOM_NAME)
+        cv = session.contentview.read(cv_name)
+        assert cv['puppet_modules']['table'][0]['Name'] == PUPPET_MODULE_CUSTOM_NAME
+        # Publish and promote CV to next environment
+        result = session.contentview.publish(cv_name)
+        assert result['Version'] == VERSION
+        result = session.contentview.promote(cv_name, VERSION, env.name)
+        assert 'Promoted to {}'.format(env.name) in result['Status']
+
+
+@tier3
+def test_positive_delete_with_kickstart_repo_and_host_group(session):
+    """Check that Content View associated with kickstart repository and
+    which is used by a host group can be removed from the system
+
+    :id: 7b076f55-72c9-4413-a592-92a47b51cb0a
+
+    :customerscenario: true
+
+    :expectedresults: Deletion was performed successfully
+
+    :BZ: 1417072
+
+    :CaseLevel: Integration
+    """
+    hg_name = gen_string('alpha')
+    sat_hostname = settings.server.hostname
+    org = entities.Organization().create()
+    # Create a new Lifecycle environment
+    lc_env = entities.LifecycleEnvironment(organization=org).create()
+
+    # Create a Product and Kickstart Repository for OS distribution content
+    product = entities.Product(organization=org).create()
+    repo = entities.Repository(product=product, url=settings.rhel6_os).create()
+
+    # Repo sync procedure
+    call_entity_method_with_timeout(repo.sync, timeout=3600)
+
```

Comment:
nitpick: imo these empty lines are redundant as well - your test contains nice comments which help to distinguish commands not worse than empty lines.